ita-social-projects/GreenCity
# 5602 bug multiple event filters

# GreenCityPR

## Issue Link :clipboard:
[[Event Dashboard] The multiple filters selected simultaniously don't display filtered Events #5602](https://github.com/ita-social-projects/GreenCity/issues/5602)

## Summary Of Changes :fire:
Added method in EventService getAllFilteredEvents() that returns filtered all events at the same time by multiple filters and sorted by id DESC.

## Added
FilterEventDto class that is event filter parameters: eventTime, status, cities, tags

## Changed
EventController method getEvent(), now it returns filtered events.
EventControllerTests changed
EventServiceImplTests added
EventService added method  getAllFilteredEvents() 
EventServiceImpl added methods to achieve filtratioan and sorting logic

## How to test :clipboard:
Use swagger: Open [events-controller](http://localhost:8080/swagger-ui.html#/events-controller/getEvent)
To test filters: 
add eventTime: FUTURE/PAST
add cities: Kyiv/Lviv.....
add statuses: OPEN/CLOSED/SUBSCRIBED/SAVED/CREATED
add tags: ECONOMIC/SOCIAL/ENVIRONMENTAL

possible to add multiple choices, but make sure that event that corresponds to all added filters exists.
